Makefile

```diff
@@ -0,0 +1,36 @@
+GO ?= go
+GH_ORG ?= Azure-Samples
+ROOT = github.com/$(GH_ORG)/azure-sdk-for-go-samples
+BASE = $(GOPATH)/src/$(ROOT)
+
+# tested on 1.9+, no need to exclude /vendor
+# to specify packages to skip: `PKGS_SKIP_RE='storage|graphrbac' make ...`
+PKGS_SKIP_RE ?= ''
+PKGS         != $(GO) list ./... | sed 's@$(ROOT)/@./@' | grep -v -E "$(PKGS_SKIP_RE)"
+
+# uses Azure resources
+test: dep
+	$(GO) test -v $(PKGS)
+
+test_pr: lint
+	$(GO) build -v \
+		$(shell $(GO) list ./... | sed 's@$(ROOT)/@./@' | grep -v -E "$(PKGS_SKIP_RE)")
```

Comment:
[nit] Shouldn't this be simplified to:

``` bash
test_pr: lint
    $(GO) build -v $(PKGS)
```

Also, why isn't it named 'build'?

Reply:
Travis didn't seem to be able to execute the script to set the variable globally, so I worked around it by putting it inline. I didn't remove the top-level script though, would like to go back to that in the future.

I named it "test" cause its purpose is to test, it does that by linting and building. I think "build" understates it, but I could break out build into a separate target in the future.